File: dms/src/automation/twitter.py
from .base import BaseDMSender
from playwright.sync_api import expect

class TwitterDMSender(BaseDMSender):
    def _login(self, page):
        page.goto("https://twitter.com/login")
        page.fill("input[autocomplete='username']", self.credentials['username'])
        page.click("div[role='button']:has-text('Next')")
        page.fill("input[name='password']", self.credentials['password'])
        page.click("div[data-testid='LoginForm_Login_Button']")
        page.wait_for_load_state("networkidle")

    def _navigate_to_dm(self, page, recipient):
        page.goto(f"https://twitter.com/messages/compose")
        page.fill("input[data-testid='searchPeople']", recipient)
        page.click(f"div[role='option']:has-text('{recipient}')")
        page.click("div[data-testid='nextButton']")
        expect(page.locator("div[data-testid='dmComposerTextInput']")).to_be_visible()

    def _send_message(self, page, message):
        page.fill("div[data-testid='dmComposerTextInput']", message)
        page.click("div[data-testid='dmComposerSendButton']")
        page.wait_for_load_state("networkidle")

File: dms/lib/twitter.ts
// lib/twitterAutomation.ts
import { chromium, Browser, Page } from 'playwright';

export async function sendTwitterDM(username: string, password: string, recipient: string, message: string) {
  const browser: Browser = await chromium.launch({ headless: false });
  const page: Page = await browser.newPage();

  try {
    // Login to Twitter
    await page.goto('https://twitter.com/login');
    await page.fill('input[name="text"]', username);
    await page.click('div[role="button"]:has-text("Next")');
    await page.fill('input[name="password"]', password);
    await page.click('div[data-testid="LoginForm_Login_Button"]');

    // Wait for login to complete
    await page.waitForNavigation();

    // Navigate to DM page
    await page.goto(`https://twitter.com/messages/compose`);
    
    // Search for recipient
    await page.fill('input[data-testid="searchPeople"]', recipient);
    await page.click(`div[role="option"]:has-text("${recipient}")`);

    // Send message
    await page.fill('div[data-testid="dmComposerTextInput"]', message);
    await page.click('div[data-testid="dmComposerSendButton"]');

    console.log(`Message sent to ${recipient} on Twitter successfully.`);
  } catch (error) {
    console.error('Error occurred:', error);
  } finally {
    await browser.close();
  }
}